### ISSUE TITLE
Communicating between JS and Native not working on iOS 

### ISSUE THREAD

**Issue by javbk201** [react-native-webview/react-native-webview#3624]
**Bug description:**

**To Reproduce:**
1. Create a new app with expo or cli.
2. Install react-native-webview
3. install pods 
4. copy and paste the code below in App.tsx

```
import { StyleSheet, View } from 'react-native'
import React from 'react'
import WebView from 'react-native-webview'

const App = () => {
  const html = `
      <html>
      <head></head>
      <body>
        <script>
          setTimeout(function () {
            window.ReactNativeWebView.postMessage("Hello!")
          }, 2000)
        </script>
      </body>
      </html>
    `;
  return (
    <View style={styles.container}>
      <WebView
        source={{html}}
        onMessage={event => {
          alert(event.nativeEvent.data);
        }}
      />
    </View>
  );
};

export default App

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
})
```

**Expected behavior:**
This code should to show an Alert after 2 seconds. But in iOS does not show it.

**Environment:**
 - OS: Mac OS Sequoia
 - OS version: 15.1
 - react-native version: 0.76.2
 - react-native-webview version: 13.12.4
 - iOS Version: 18.0

**Comment by mehmetsalihyaldiz** [comment]
same here

**Comment by mlazari** [comment]
> If source={{ html }}, iOS is fine

@mehmetsalihyaldiz As I understand from the provided example, this is the problematic case actually - html source on iOS.

I see the same issue. My WebView is in a Modal. First time when I open the modal it doesn't work on iOS. The second and next times already works.

I wrote a small repro similar to my case here: https://snack.expo.dev/@mlazari/webview-repro?platform=ios

As you can see the first time I open the modal, the alert is not shown:
https://github.com/user-attachments/assets/ca70c88d-f8e1-42f7-b3bf-98fd5b383a25

**Comment by mlazari** [comment]
Downgrading to 13.12.2 fixes the issue for me. It's strange that in my expo snack it's also 13.12.2, but maybe the Expo Go app that I have installed uses a newer version (I am not sure how dependency versions work with Expo Go).

**Comment by mlazari** [comment]
This [patch](https://www.npmjs.com/package/patch-package) for 13.12.4 works for me too:
https://github.com/user-attachments/files/17937022/react-native-webview%2B13.12.4.patch
```
diff --git a/node_modules/react-native-webview/apple/RNCWebViewImpl.m b/node_modules/react-native-webview/apple/RNCWebViewImpl.m
index 97d9780..16704c2 100644
--- a/node_modules/react-native-webview/apple/RNCWebViewImpl.m
+++ b/node_modules/react-native-webview/apple/RNCWebViewImpl.m
@@ -565,14 +565,14 @@ - (void)initializeWebView
 }
 
 // react-native-mac os does not support didMoveToSuperView
-#if !TARGET_OS_OSX
-- (void)didMoveToSuperview
-{
-  if (self.superview != nil && _webView == nil) {
-    [self initializeWebView];
-  }
-}
-#endif // !TARGET_OS_OSX
+// #if !TARGET_OS_OSX
+// - (void)didMoveToSuperview
+// {
+//   if (self.superview != nil && _webView == nil) {
+//     [self initializeWebView];
+//   }
+// }
+// #endif // !TARGET_OS_OSX
 
 - (void)didMoveToWindow
 {
```

**Comment by pmprete** [comment]
This is how i fixed it
"react-native-web": "~0.19.12",
"react-native-webview": "13.12.2",

const [ ready, setReady ] = useState(false);
const [ webkey, setWebkey ] = useState(0);
<Webview
  key = { webkey }

  onLoadEnd={()=>{
    if( !ready ){
     setWebKey(Date.now());
       setReady(true)
   }
  }}
/>

just change the key state in onLoadEnd, It can avoid multiple redraws

Source: https://github.com/react-native-webview/react-native-webview/issues/3616#issuecomment-2494300117

**Comment by mlazari** [comment]
@pmprete Since you are already using the older version 13.12.2, I don't think you need that workaround, just make sure to reinstall pods and rebuild the app after installing 13.12.2, that ensures you are using the WebView's native code from 13.12.2.

**Comment by mlazari** [comment]
Anyone willing to test a fix for this issue, please try these steps:
- replace `"react-native-webview": "13.12.4"` with `"react-native-webview": "github:react-native-webview/react-native-webview#573ce151683296f469de74fb21aa6de8c58d1dce"` in your package.json file
- install npm dependencies with `npm install` / `yarn install` (depending on which package manager you are using)
- reinstall pods (`pod install` in the `ios` folder) or if you are using expo run `npx expo prebuild`
- rebuild the app (e.g with `npx react-native run-ios` or if you are using expo with `npx expo run:ios`; or from Xcode)
- check if this issue is fixed

**Comment by mlazari** [comment]
This should be fixed in version 13.12.5 by this PR: https://github.com/react-native-webview/react-native-webview/pull/3615

**Comment by pmprete** [comment]
Worked for me in version 13.12.5  thanks @mlazari
